timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/LeaseExpirationTimer.java

```diff
@@ -15,26 +15,33 @@
  */
 package com.palantir.atlasdb.timelock.lock;
 
-import com.palantir.common.time.Clock;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+import com.palantir.common.time.NanoTime;
 
 public class LeaseExpirationTimer {
 
-    public static final long LEASE_TIMEOUT_MILLIS = 20_000;
+    public static final Duration LEASE_TIMEOUT = Duration.ofMillis(20_000);
 
-    private volatile long lastRefreshTimeMillis;
-    private final Clock clock;
+    private final AtomicReference<NanoTime> lastRefreshTime;
+    private final Supplier<NanoTime> clock;
 
-    public LeaseExpirationTimer(Clock clock) {
+    public LeaseExpirationTimer(Supplier<NanoTime> clock) {
         this.clock = clock;
-        this.lastRefreshTimeMillis = clock.getTimeMillis();
+        this.lastRefreshTime = new AtomicReference<>(clock.get());
```

Comment:
given that this uses no atomic operations (cas, etc) might be worth just leaving this as volatile?